src/components/navigation.ts

```diff
@@ -1,31 +1,47 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnDestroy } from '@angular/core';
 import { Services } from '../services';
 import { TraversingComponent } from '../traversing';
 import { NavTree } from '../interfaces';
 import { Target } from 'angular-traversal'
+import { Subscription } from 'rxjs/Subscription';
+import { Observable } from 'rxjs/Observable';
 
 @Component({
   selector: 'plone-navigation',
-  template: `<plone-navigation-level
+  template: `
+    <plone-navigation-level
       [links]="links"></plone-navigation-level>`
 })
-export class Navigation extends TraversingComponent {
+export class Navigation extends TraversingComponent implements OnDestroy {
 
   @Input() root: string = '/';
   @Input() depth: number = -1;
   links: NavTree[];
+  refreshNavigation: Subscription;
 
-  constructor(
-    public services: Services,
-  ) {
+  constructor(public services: Services,) {
     super(services);
   }
 
   onTraverse(target: Target) {
-    this.services.navigation
-      .getNavigationFor(target.context['@id'], this.root, this.depth)
+    const component = this;
+    const navigation = component.services.navigation;
+    component.removeSubscriptions();
+    this.refreshNavigation = Observable.of(null)
```

Comment:
`Observable.of(null)` funny :), never saw that before. Why can't we just start with `navigation.refreshNavigation` and then mergeMap?